validator/sawtooth_validator/execution/scheduler_serial.py

```diff
@@ -49,6 +49,9 @@ def __init__(self, squash_handler, first_state_hash):
         self._last_in_batch = []
         self._last_state_hash = first_state_hash
 
+    def __del__(self):
```

Comment:
The scheduler is holding the execution of a thread in the thread pool. Since the condition held by the SerialScheduler must be signaled for the thread to be reclaimed. The penalty for not doing this is the lose of a thread in the Execution pools, so it is better to error on the side of caution cleaning up these resources then let these resources leak.